### PR TITLE
run-onnx-lib script and docs fixes

### PR DESCRIPTION
**Pull request by sorenlassen**
fixed some issues with the build-run-onnx-lib.sh tool and its documentation

edited the example commands in the docs to call the script with `sh`

clarified Mac issue with running statically linked version in the directory where model library was built, and explained how to use `otool -L` to find directory to run from in case of static linking on Mac

always build with -std=c++17

allow llvm-project location to be set with LLVM_PROJECT, otherwise read from $MLIR_DIR if defined

removed some duplication

moved usage string up to the top of RunONNXLib.cpp, so it's visible next to the top of file block comment, and removed the duplicate usage comment at the top which was out of date

### CODE REVIEW

File: docs/Testing.md [hunk 1]
```diff
@@ -151,15 +151,20 @@ We first need to compile the tool, which can be done in one of two modes.
...
-(cd test/backend; run-onnx-lib)
+sh ../utils/build-run-onnx-lib.sh test/backend/test_add/test_add.so
+# Run the tool to run the model.
+Debug/bin/run-onnx-lib
+# or, on Mac, run the tool in the directory where the model was built
+(cd test/backend; ../../Debug/bin/run-onnx-lib)
+# if test_add.so was built in `test/backend`:
+cd test/backend; ../../Debug/bin/onnx-mlir --EmitLib test_add/test_add.onnx
 ```
+(You can see the path of the library with `otool -L test_add.so` on Mac.)
```
Comment (sorenlassen) [inline docs/Testing.md:167]: I tested on Mac and Linux

on Linux things worked fine without changing directory, so I'm guessing that this is a Mac specific problem and I edited the explanation accordingly, and added the Mac-specific `otool` advice

please let me know if anyone is aware of this problem existing on other platforms than Mac

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, would be great to hear from Window folks if there are updates to the scripts. I suspect there is, not sure if one script can accommodate both, if that is the case, then let's have this one working for Mac/Linux and a subsequent PR can add window support as needed

File: docs/Testing.md [hunk 2]
```diff
@@ -169,9 +174,9 @@ any directories as long as you pass the `.so` model file at runtime to the tool.
 ``` sh
 # Compile tool without a model.
 cd onnx-mlir/build
-. ../utils/build-run-onnx-lib.sh
+sh ../utils/build-run-onnx-lib.sh
 # Run the tool with an argument pointing to the model.
-run-onnx-lib test/backend/test_add.so
+Debug/bin/run-onnx-lib test/backend/test_add/test_add.so
```
[the remark this answers is not in the document]
Comment (sorenlassen) [reply docs/Testing.md:179]: added

**Comment by sorenlassen** [comment]
@AlexandreEichenberger how about we merge this? I can quickly roll it back if it proves to be a problem for any WIndows folks

**Comment by AlexandreEichenberger** [comment]
> how about we merge this? 

will do

**Pull request merged**